### PR TITLE
kind: deploy local volume provisioner

### PR DESCRIPTION
**Pull request by maiqueb**
Kind default StorageClass is named 'standard', while kubevirt
func tests assume it to be called 'local' - as is for all the
other providers.
    
To enable the kind-k8s-1.17-ipv6 provider to work out of the box,
the `local-volume-provisioner` daemonset must be deployed. It will
take care of:
  - create the 'local' storage class
  - discover all mount points in its configuration path (via config
    map) and expose those mount points as PersistentVolumes.
    
For more information, check the local static provisioner repo,
located at [0].
    
This will enable the functional_tests labeled as 'PVC' to be
executed.
    
[0] - https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner

Signed-off-by: Miguel Duarte Barroso <mdbarroso@redhat.com>

### CODE REVIEW

File: cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 set -e
 
+DOCKER="${CONTAINER_RUNTIME:-docker}"
```
Comment (qinqon) [inline cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: This is unrelated

File: cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh [hunk 2]
```diff
@@ -11,4 +13,24 @@ source ${KUBEVIRTCI_PATH}/cluster/kind/common.sh
 function up() {
     cp $KIND_MANIFESTS_DIR/kind-ipv6.yaml ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/kind.yaml
     kind_up
+    # remove the rancher.io kind default storageClass
+    _kubectl delete sc standard
+
+    nodes=$(_kubectl get nodes -o=custom-columns=:.metadata.name | awk NF)
+    for node in $nodes; do
+        # Create local-volume directories
+        for i in {1..10}; do
```
Comment (qinqon) [inline cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: why 10 ?

File: cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh [hunk 3]
```diff
@@ -11,4 +13,24 @@ source ${KUBEVIRTCI_PATH}/cluster/kind/common.sh
...
+        done
+        $DOCKER exec -it $node bash -c "chmod -R 777 /var/local/kubevirt-storage/local-volume"
+    done
+
+    # create the `local` storage class - which functional tests assume to exist
+    _kubectl apply -f $KIND_MANIFESTS_DIR/local-volume.yaml
+}
+
+function mount_disk() {
+    $DOCKER exec -it $node bash -c "mkdir -p /var/local/kubevirt-storage/local-volume/disk${i}"
```
Comment (qinqon) [inline cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: pass index as argument
```bash
local idx=$1
```

File: cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh (hunk 2 above)
Comment (awels) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: 10 is basically a number that seemed large enough to give enough PVs for most tests at the time. With the ability to add more later, it seemed good enough. I still haven't found any cases where 10 was not more than enough.

File: cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh [hunk 4]
```diff
@@ -11,4 +13,25 @@ source ${KUBEVIRTCI_PATH}/cluster/kind/common.sh
...
+    for node in $nodes; do
+        # Create local-volume directories
+        for i in {1..10}; do
+            node=$node mount_disk $i
+        done
+        $DOCKER exec -it $node bash -c "chmod -R 777 /var/local/kubevirt-storage/local-volume"
+    done
+
+    # create the `local` storage class - which functional tests assume to exist
+    _kubectl apply -f $KIND_MANIFESTS_DIR/local-volume.yaml
```
Comment (awels) [inline cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: Any reason not to use the existing one in cluster-provision/manifests/local-volume.yaml

Comment (awels) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: So from what I understand, there is a provision step that builds the images, and in that step that file is copied onto the nodes. And during the cluster startup the yaml is applied from the node. But this is in the VM case, I guess kind is different?

Comment (maiqueb) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: Yes, seems we just fetch the `kind` that k8s-sigs provides: https://github.com/kubevirt/kubevirtci/blob/master/cluster-up/cluster/kind/common.sh#L35

Comment (awels) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: Interesting. Well I simply want to avoid having the same file in 2 places so we don't over time get drift between the two. I don't care that much where it lives.

File: cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh (hunk 2 above)
Comment (qinqon) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: Can you add q comment about it ?

Comment (qinqon) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: Can you add q comment about it ?

Comment (qinqon) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: Can you add q comment about it ?

File: cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh (hunk 1 above)
Comment (qinqon) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: Fair enough

File: cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh (hunk 4 above)
Comment (qinqon) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: At kubevirtci we are doing transition on having providers isoletad, for easy debugging and releasing, even if it means duplication, in case of running needed is much better, and we don't have that much.

File: cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh (hunk 2 above)
Comment (maiqueb) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: This is actually public facing information, available in the [kubevirtci storage README](https://github.com/kubevirt/kubevirtci/blob/master/cluster-provision/STORAGE.md#storage-providers-inside-the-generated-images) - "By default, we have pre-provisioned 10 local volumes per node, which are available to make requests against. You can manually bind mount more directories to increase this number.".

Added that link to the already existing comment, hope that's enough.

Comment (qinqon) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: good

File: cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh [hunk 5]
```diff
@@ -11,4 +13,26 @@ source ${KUBEVIRTCI_PATH}/cluster/kind/common.sh
...
+}
+
+function mount_disk() {
+    local idx=$1
+    $DOCKER exec -it $node bash -c "mkdir -p /var/local/kubevirt-storage/local-volume/disk${idx}"
```
Comment (qinqon) [inline cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: Same for node
local node=$2 so it's easier to follow.

Comment (maiqueb) [reply cluster-up/cluster/kind-k8s-1.17.0-ipv6/provider.sh]: Done.
